GitHubReferences.ps1

```diff
@@ -0,0 +1,442 @@
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OwnerName
+        Owner of the repository.
+        If not supplied here, the DefaultOwnerName configuration property value will be used.
+
+    .PARAMETER RepositoryName
+        Name of the repository.
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER Reference
+        The name of the reference to be deleted (eg: heads/master)
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        Remove-GitHubReference -OwnerName microsoft -RepositoryName PowerShellForGitHub -Reference heads/master
+    #>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName='Elements')]
+    [Alias('Delete-GitHubReference')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName='Uri')]
+        [string] $Uri,
+
+        [Parameter(Mandatory)]
+        [string] $Reference,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog -Invocation $MyInvocation
+
+    $elements = Resolve-RepositoryElements -BoundParameters $PSBoundParameters -DisableValidation
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    if ($OwnerName -xor $RepositoryName)
+    {
+        $message = 'You must specify both Owner Name and Repository Name.'
+        Write-Log -Message $message -Level Error
+        throw $message
+    }
+
+    $uriFragment = "repos/$OwnerName/$RepositoryName/git/refs/$Reference"
+    $description =  "Deleting Reference $Reference from repository $RepositoryName"
+
+    $params = @{

```

Comment:
A recent PR introduced the idea of getting user confirmation on delete actions (which can be overridden by a commandline parameter).  I think it's a pretty solid idea, and I think we should apply that approach on all new "delete" methods.   Take a [look here](https://github.com/microsoft/PowerShellForGitHub/commit/536b762425d51a181166c2c47ad2b00014911d1d#diff-2a0b254e60269a3a0794d63c7922522dR518-R549) for more info.

I'd ask that you _also_ add an additional Example to here that shows how to avoid the prompt (as that reference that I pointed to does as well).

Reply:
Agreed. That looks like a good idea. 
suggestion: Can we have the flag as -NoConfirm or similar so that the user need not type in -Confirm:$false ?